include/sound/sof.h

```diff
@@ -92,9 +92,7 @@ static inline int sof_mach_get_mach_size(const struct snd_soc_sof_mach *mach)
  */
```

Comment:
overall comment for this patch: there are too many changes here, it's a bit too hard to review. We should try and split in steps if possible